var/spack/repos/builtin/packages/sw4/package.py

```diff
@@ -8,59 +8,66 @@
 
 
 class Sw4(MakefilePackage):
-    """This package builds SW4 with MPI, OpenMP, HDF5, FFTW, PROJ, and ZFP."""
+    """SW4 implements substantial capabilities for 3-D seismic modeling."""
 
     homepage = "https://github.com/geodynamics/sw4"
     git = "https://github.com/geodynamics/sw4.git"
 
     maintainers("houjun", "andersp")
 
+    license("GPL-2.0-or-later")
+
     version("master", branch="master")
     version("developer", branch="developer")
     version("3.0", tag="v3.0", commit="13e6d431976f7fc49124c997bf87353aa7afd35e")
 
-    depends_on("cxx", type="build")  # generated
-    depends_on("fortran", type="build")  # generated
-
-    variant("openmp", default=True, description="build with OpenMP")
-    variant("hdf5", default=True, description="build with HDF5")
-    variant("proj", default=True, description="build with proj")
-    variant("zfp", default=False, description="build with ZFP")
-    variant("fftw", default=True, description="build with FFTW")
+    variant("openmp", default=True, description="Build with OpenMP")
+    variant("proj", default=True, description="Build with PROJ")
+    variant("hdf5", default=True, description="Build with HDF5")
+    variant("zfp", default=False, when="+hdf5", description="Build with ZFP")
+    variant("fftw", default=True, description="Build with FFTW")
+    variant("debug", default=False, description="Build with debugging symbols")
 
+    depends_on("cxx", type="build")
+    depends_on("fortran", type="build")
     depends_on("mpi")
     depends_on("blas")
     depends_on("lapack")
-    depends_on("proj", when="+proj")
-    depends_on("hdf5+mpi", when="+hdf5")
-    depends_on("fftw@3.0: +mpi", when="+fftw")
+    depends_on("python")
+    depends_on("proj@9:", when="+proj")
+    depends_on("hdf5@1.14: +mpi", when="+hdf5")
+    depends_on("py-h5py", when="+hdf5")
     depends_on("zfp", when="+zfp")
     depends_on("h5z-zfp@develop", when="+zfp")
-    depends_on("python")
-    depends_on("py-h5py")
+    depends_on("fftw@3: +mpi", when="+fftw")
     depends_on("llvm-openmp", when="%apple-clang +openmp")
 
     def edit(self, spec, prefix):
```

Comment:
Instead of `edit`, it may make more sense to define a `setup_build_emvironment` function here.

Reply:
Sure, I changed it to `setup_build_environment`.